### PR TITLE
Updated typespec common script

### PR DESCRIPTION
**Pull request by raych1**
- Refine regex
- Remove temp tspconfig.yaml
- Use relative path as directory in tsplocation.yaml

@weshaggard , the third change is a fix of a bug and needs to get merged to unblock integration work with other languages.

### CODE REVIEW

File: eng/common/scripts/TypeSpec-Project-Process.ps1 [hunk 1]
```diff
@@ -124,8 +124,20 @@ if ($TypeSpecProjectDirectory -match '^https://github.com/(?<repo>Azure/azure-re
...
+    $TypeSpecProjectDirectory = $Matches["path"]
+  } else {
+    Write-Error "'$TypeSpecProjectDirectory' doesn't have 'specification' in path."
+    exit 1
+  }
 }
+
 $tspConfigYaml = Get-Content $tspConfigPath -Raw | ConvertFrom-Yaml
+
+# delete the tmp tspconfig.yaml downloaded from github
```
Comment (konrad-jamrozik) [inline eng/common/scripts/TypeSpec-Project-Process.ps1:137]:
```suggestion
# delete the temporary tspconfig.yaml downloaded from github
```

File: eng/common/scripts/TypeSpec-Project-Process.ps1 [hunk 2]
```diff
@@ -124,8 +124,20 @@ if ($TypeSpecProjectDirectory -match '^https://github.com/(?<repo>Azure/azure-re
     Write-Error "Failed to find tspconfig.yaml in '$TypeSpecProjectDirectory'"
     exit 1
   }
+  if($TypeSpecProjectDirectory -match "^.*/(?<path>specification/.*)$") {
```
Comment (weshaggard) [inline eng/common/scripts/TypeSpec-Project-Process.ps1:127]: We should move to before line 122 where you are using it to find the tspconfig location. If this path isn't valid we should know it there.

**Review by weshaggard — APPROVED** [review]
One suggestion otherwise looks good.

File: eng/common/scripts/TypeSpec-Project-Process.ps1 (hunk 2 above)
Comment (raych1) [reply eng/common/scripts/TypeSpec-Project-Process.ps1:127]: Updated.

**Comment by weshaggard** [comment]
@raych1 you did not follow the workflow process again. See directions at https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/README.md#workflow. This is why your checks in your current PR are failing https://github.com/Azure/azure-sdk-tools/pull/6250

**Comment by raych1** [comment]
> @raych1 you did not follow the workflow process again. See directions at https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/README.md#workflow. This is why your checks in your current PR are failing #6250

@weshaggard , sorry, I wasn't aware of this process. I found you helped on the previous PR sync to language repo. I just triggered the sync pipeline above and only one PR failed by other reason then I manually merge it.

**Pull request merged**